Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -400,6 +400,19 @@ class MessageDTO: NSManagedObject {
         request.predicate = NSPredicate(format: "readMessagesFromCurrentUser.id CONTAINS %@", messageId)
         return (try? context.count(for: request)) ?? 0
     }
+    
+    static func loadLastMessage(from userId: String, in cid: String, context: NSManagedObjectContext) -> MessageDTO? {
+        let request = NSFetchRequest<MessageDTO>(entityName: entityName)
+        request.predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [
+            channelPredicate(with: cid),
+            .init(format: "user.id == %@", userId),
+            .init(format: "type != %@", MessageType.ephemeral.rawValue),
+            .init(format: "localMessageStateRaw == nil")
```

Comment:
nit: Would be nice to extract this one to `messageSentPredicate()` since it is hard to understand what is this doing. And use this in the other places where we have this duplicated as well 👍 
```suggestion
            .init(format: "localMessageStateRaw == nil")
```

Reply:
Done